### PR TITLE
Introduce `flex-list` & `flex-item` elements for Gitea UI

### PR DESCRIPTION
**Pull request by denyskon**
This PR introduces a new UI element type for Gitea called `flex-item`. It consists of a horizontal card with a leading, main and trailing part:

grafik

The idea behind it is that in Gitea UI, we have many cases where we use this kind of layout, but it is achieved in many different ways:
  - grid layout
  - `.ui.list` with additional hacky flexbox
  - `.ui.key.list` - looks to me like a style set originally created for ssh/gpg key list, was used in many other places
  - `.issue.list` - created for issue cards, used in many other places
  - ...

This new style is based on `.issue.list`, specifically the refactoring of it done in #25750.

In this PR, the new element is introduced and lots of templates are being refactored to use that style. This allows to remove a lot of page-specific css, makes many of the elements responsive or simply provides a cleaner/better-looking way to present information.

A devtest section with the new style is also available.

<details>
<summary>Screenshots (left: before, right: after)</summary>

Bildschirmfoto vom 2023-07-09 21-01-21
Bildschirmfoto vom 2023-07-09 21-01-56
Bildschirmfoto vom 2023-07-09 21-02-45
Bildschirmfoto vom 2023-07-09 21-03-44
Bildschirmfoto vom 2023-07-09 21-04-52
Bildschirmfoto vom 2023-07-09 21-05-25
Bildschirmfoto vom 2023-07-09 21-06-35
Bildschirmfoto vom 2023-07-09 21-09-03
Bildschirmfoto vom 2023-07-09 21-09-44
Bildschirmfoto vom 2023-07-09 21-10-27
Bildschirmfoto vom 2023-07-09 21-11-12
Bildschirmfoto vom 2023-07-09 21-12-01
Bildschirmfoto vom 2023-07-09 21-17-44
Bildschirmfoto vom 2023-07-09 21-18-27
Bildschirmfoto vom 2023-07-09 21-19-18
Bildschirmfoto vom 2023-07-09 21-29-13
Bildschirmfoto vom 2023-07-09 21-30-11
Bildschirmfoto vom 2023-07-09 21-32-44
Bildschirmfoto vom 2023-07-09 21-33-28

</details>

### CODE REVIEW

**Comment by KN4CK3R** [comment]
Is there a way to reduce the space above "yesterday" at the bottom?
https://user-images.githubusercontent.com/47871822/252165284-c1b7c22e-3e5a-46d4-b8d6-5560db478c0b.png

**Comment by denyskon** [comment]
Done :)
grafik

**Comment by wxiaoguang** [comment]
IIRC there is already a `flex-items-block` ?

File: templates/devtest/gitea-ui.tmpl
Comment (wxiaoguang) [inline templates/devtest/gitea-ui.tmpl]: For most frameworks, when there is a "list", then the "children elements" are called "item" 🤔 

"card" seems an uncommon concept and it conflicts with Fomantic UI card component.

Comment (puni9869) [reply templates/devtest/gitea-ui.tmpl]: Agreed with @wxiaoguang

Comment (denyskon) [reply templates/devtest/gitea-ui.tmpl]: Renamed `flex-card` to `flex-item`

**Comment by silverwind** [comment]
Missed this one, will review soon.

**Comment by silverwind** [comment]
Could you resolve conflicts once more?

**Comment by denyskon** [comment]
@silverwind Done

**Comment by puni9869** [comment]
I would like to appreciate this PR @denyskon 💯 . A request whenevery you are free, could you write a small an example for `flex-list` & `flex-item` in  our contribution guide for frontend. Let make it a standard. Let make our contribution guide better and develoer friendly.
I don't want to block this PR, can be done in next PR.

**Comment by denyskon** [comment]
I thought devtest is a better place for it.... Maybe let's wait for what other maintainers think?

**Comment by puni9869** [comment]
> I thought devtest is a better place for it.... Maybe let's wait for what other maintainers think?

Yeah thats a great place, but is `devtest` is mentioned in frontend contribution guide?

**Comment by silverwind** [comment]
Yes, devtest is for any UI live example.

**Comment by denyskon** [comment]
@puni9869 
-> https://github.com/go-gitea/gitea/pull/26249

**Comment by puni9869** [comment]
Let's roll this Pr. If it is finished.

**Comment by denyskon** [comment]
> Let's roll this Pr. If it is finished.

Zero approvals so far 🥲

**Review by puni9869 — APPROVED** [review]

**Review by wxiaoguang — APPROVED** [review]
Haven't gotten time for testing it.

Some selectors are likely having unnecessary dependency, eg: `.flex-list .branches .truncated-name` and `flex-list .branches .branch`. Ideally, the "flex-list.css" is a independent layout framework, the "page" related selectors may need a totally new selector, eg: `.xxx-branch-list .branch ...`.

But I can vote my LGTM, while I think some improvements could be done later.

**Comment by puni9869** [comment]
> Haven't gotten time for testing it.
> 
> Some selectors are likely having unnecessary dependency, eg: `.flex-list .branches .truncated-name` and `flex-list .branches .branch`. Ideally, the "flex-list.css" is a independent layout framework, the "page" related selectors may need a totally new selector, eg: `.xxx-branch-list .branch ...`.
> 
> But I can vote my LGTM, while I think some improvements could be done later.

Correct, that can be done in follow up pr and minor suggestion I also have. Not blotting this PR.

**Comment by puni9869** [comment]
image

**Comment by denyskon** [comment]
@wxiaoguang Thanks for noticing, I moved the unrelated styles to the issue-list css file where they belong.

**Comment by denyskon** [comment]
I would like to wait for silverwind's opinion before merging, as this is a large refactoring.

**Review by silverwind — APPROVED** [review]

**Comment by silverwind** [comment]
Looks good and I can't really spot any difference between the old issuelist, so let's give this a try.

File: templates/org/member/members.tmpl
Comment (Sonic853) [inline templates/org/member/members.tmpl:2]: Why to delete this "members" class?

Comment (puni9869) [reply templates/org/member/members.tmpl:2]: Is there any impact? of regression

Comment (silverwind) [reply templates/org/member/members.tmpl:2]: Maybe they were depending on this class for customization. I wonder if there would be a better way to indicate the current page to CSS, like an data attribute that indicates the matching template file or router. We could make better stability guarantees for those then these flimsy classes.

Comment (Sonic853) [reply templates/org/member/members.tmpl:2]: > Is there any impact? of regression

Although there is currently no impact. But there is no way to tell under Vanilla JS whether it is a "Members" page.

Comment (silverwind) [reply templates/org/member/members.tmpl:2]: In JS, you could check the URL which I guess is more stable than these classes. e.g. check if `/^\/[^/]+/[^/]+/members$/.test(window.location.pathname)`.

**Pull request merged**
